### ISSUE TITLE
Tuple errors when initializing record with param of `uint`

### ISSUE THREAD

**Issue by BryantLam** [chapel-lang/chapel#12462]
Related #12461

Bug. Low priority since there's a workaround. Tuple problems with initializing a generic record when that record contains param of `uint` type.

```chapel

record D1 {
  type t;
  var data: t;
}

record D1Array {
  type t;
  var data: [1..10] D1(t);
}

record D2 {
  param p: int;
  
  type t;
  var data: p*t;
}

record D2Array {
  param p: int;
  
  type t;
  var data: [1..10] D2(p, t);
}

record R {
  type ut;
  param u: ut;
}

proc main() {
  // Ok
  D1Array(R(int, 42));
  D1Array(R(uint, 42:uint));
  
  // error: Default expression not convertible to formal type
  D1Array(R(uint, 42));
  
  
  // Ok
  D2Array(2, R(int, 42));
  D2Array(2, R(uint, 42:uint));
  
  // error: cannot assign expression of type R(uint(64),42:uint(64)) to field 'x1' of type R(uint(64),42)
  D2Array(2, R(uint,42));
}
```
`chpl version 1.19.0 pre-release`

**Comment by BryantLam** [comment]
Err, `D1Array` doesn't have a tuple in it, but you get the idea.

**Comment by benharsh** [comment]
I believe I have a fix for this, but it won't be in the 1.19 release.

**Comment by BryantLam** [comment]
Brad suggested this code is related:
```chapel
record R { param size: uint; }
var r: R(64) = new R(64)
```
```terminal
error: unresolved call 'R(64:uint(64)).init=(R(64:uint(64)))'
note: candidates are: R.init=(other: this.type)
```
Notably, the error goes away if I actually define the same R.init inside:
```chapel
record R {
  param size: uint;
  proc init=(other: this.type) {}
}
```
